### PR TITLE
Enhancement 4790 order prioritization logic

### PR DESCRIPTION
**Pull request by Hemadri-Chinna**
## Requirements

https://github.com/pucardotorg/dristi/issues/4790

- [x] This PR has a proper title that briefly describes the work done
- [x] Please ensure the branch name follows naming convention - feature-githubissunumber-xxx, bug-githubissunumber-xxx, enhance-githubissunumber-xxx.
- [x] I have referenced the  github issues('s)
- [x] I performed a self-review of my own code
- [x] New and existing unit tests pass locally with my changes
- [x] I have added proper logs and comments for the developed code
- [x] If this PR includes MDMS, workflow data, or deployment configuration changes:
  - [x] I have added MDMS data changes to `support/release-<release-number>-<issue-number>-mdms.json`
  - [ ] I have added workflow data changes to `support/release-<release-number>-<issue-number>-workflow.json`
  - [ ] I have added deployment configuration steps to `support/release-<release-number>/deployment.md`

## Summary by CodeRabbit

* **Improvements**
  * Better prioritization of hearing vs. order entries for more accurate published case statuses; added an API to update cases without triggering workflow.
* **Reliability**
  * Enhanced error handling and logging around hearing extraction, priority resolution and composite processing.
* **Configuration**
  * Added process-handler and stage/substage backup support plus schema/data updates to enable prioritized, entity-aware status flows.
* **Tests**
  * New unit tests for priority selection, composite items, hearing extraction and edge cases.
* **Bug Fix**
  * Adjusted LPR transition to avoid unintended stage/substage backups during enrichment.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: backend/analytics/src/main/java/org/pucar/dristi/util/CaseOverallStatusUtil.java [hunk 1]
```diff
@@ -98,6 +96,25 @@ private Object processOrderOverallStatus(JSONObject request, String referenceId,
         Object orderObject = orderUtil.getOrder(request, referenceId, config.getStateLevelTenantId());
         String filingNumber = JsonPath.read(orderObject.toString(), FILING_NUMBER_PATH);
         String orderCategory = JsonPath.read(orderObject.toString(), ORDER_CATEGORY_PATH);
+		boolean isHearingFound = false;
+		TreeMap<Integer, CaseOverallStatus> priorityMap = new TreeMap<>();
+		try {
+			String hearingType = JsonPath.read(orderObject.toString(), PURPOSE_OF_NEXT_HEARING_PATH);
+			isHearingFound = hearingType != null;
+			for (CaseOverallStatusType caseOverallStatusType : caseOverallStatusTypeList) {
```
Comment (manimaarans) [inline backend/analytics/src/main/java/org/pucar/dristi/util/CaseOverallStatusUtil.java]: extract this into separate method @Hemadri-Chinna
